common/build-style/zig-build.sh

```diff
@@ -0,0 +1,41 @@
+do_build() {
+	local zig_target
+	local zig_cpu
+	case $XBPS_TARGET_MACHINE in
+		aarch64-musl|i686-musl|x86_64-musl)
+			zig_target="${XBPS_TARGET_MACHINE%-musl}-linux-musl" zig_cpu="baseline";;
+		aarch64|i686|x86_64)
+			zig_target="${XBPS_TARGET_MACHINE}-linux-gnu" zig_cpu="baseline";;
+		armv6l-musl) zig_target="arm-linux-musl" zig_cpu="generic+v6";;
+		armv6l) zig_target="arm-linux-gnu" zig_cpu="generic+v6";;
+		armv7l-musl) zig_target="arm-linux-musl" zig_cpu="generic+v7a";;
+		armv7l) zig_target="arm-linux-gnu" zig_cpu="generic+v7a";;
+		*) broken="TODO: support more target machines for the zig build style";;
```

Comment:
Why not add ppc already? Not supported upstream yet or just not implemented?

Reply:
I was just being lazy. Upstream zig has at least some level of support for everything xbps can target. Added the ppc variants to this switch and will add others when I have time though I don't have a good way to test them.